packages/core/src/esdb.ts

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright © 2020 Anticrm Platform Contributors.
+// 
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import type { Storage } from './storage'
+import { TxProcessor } from './storage'
+import type { Emb, Doc, Ref, Class, Data, Collection } from './classes'
+import type { TxAddCollection, TxCreateObject } from './tx'
+import type { Hierarchy } from './hierarchy'
+import { Client, RequestParams } from '@elastic/elasticsearch'
+import { generateId } from './utils'
+
+export class EsDb extends TxProcessor implements Storage {
+  private readonly hierarchy: Hierarchy
+  private readonly client = new Client({
```

Comment:
I propose to client connection details to EsDB and rename it to ElasticStorage. 
Assume we potentially could use different Elastic storages for different domains.